openmdao/core/problem.py

```diff
@@ -1824,7 +1825,7 @@ def check_config(self, logger=None, checks=None, out_file='openmdao_checks.out')
 
 def _assemble_derivative_data(derivative_data, rel_error_tol, abs_error_tol, out_stream,
                               compact_print, system_list, global_options, totals=False,
-                              indep_key=None, print_reverse=False,
+                              linear_constraints=None, indep_key=None, print_reverse=False,
```

Comment:
linear_constraints isn't in the doc string here.  It's a bit confusing.  Above you pass it self.driver._cons, does that include the nonlinears as well?